### PR TITLE
solution for issue #3 - buffer dynamic allocation

### PR DESCRIPTION
**Pull request by XakerTwo**
well it's pretty huge change with a lot of reworks

what was done in short

for sure, buffer now is allocated after calculations(the measurement run)
the measurement run will not reach `append_to_buffer()` (btw this function is only for the write run)
implement cache for OIDs and node lengths, related functions also was improved
note that cache for node lengths named the blind because it has no any context nor checks, it's just a stream of lengths and if something between its construction and use are changed - kaboom
files now is one-way linked list that's allocated dynamically (this change is a side effect of hwids improvement)
few improvements, like SHA1 conversions and OID encoding (in addition to cache)
few small fixes and adjustments, like encoding length in proper 2 bytes and order of attributes

well, i tested it and compared with Inf2Cat output and look like all OK

however delay was due to C point me on its difference from C++ ...
in short i was not aware about some syntax differences, but at now each change tested twice - in C (migwin) and C++(ms) editions :sweat_smile: but there can be some traces of pluses

### CODE REVIEW

**Comment by johannesthoma** [comment]
Hi [XakerTwo](https://github.com/XakerTwo) thanks a LOT for your contribution. I will
go over it in the next days, but the only thing that really matters to me is that it works
with my WinDRBD build process (which should be easy to test), so I am confident that
this can be merged as it is. Will keep you posted (sorry out sick this week)...
